src/coreclr/System.Private.CoreLib/src/System/Object.CoreCLR.cs

```diff
@@ -3,15 +3,32 @@
 
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace System
 {
     public partial class Object
     {
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "ObjectNative_GetTypeSlow")]
+        private static unsafe partial void GetTypeSlow(MethodTable* methodTable, ObjectHandleOnStack ret);
+
         // Returns a Type object which represent this object instance.
         [Intrinsic]
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public extern Type GetType();
+        public unsafe Type GetType()
```

Comment:
If it's not inlined, then it becomes an indirect call (while previously it used to be a direct call)

if it's inlined (shouldn't happen today I guess due to small callsite size vs a lot of unknown call inside of - it will scare the inliner) - it might ruin some jit opts where jit treats GetType as a 
special intrinsic and relies on it even in late phases.

We can optimize 1) if it becomes an issue on VM side (tell jit that certain managed methods can be direct call)